usr.bin/grep/grep.c

```diff
@@ -62,9 +62,8 @@ const char	*errstr[] = {
 /* 8*/	"%s (BSD grep, GNU compatible) %s\n",
 };
 
-/* Flags passed to regcomp() and regexec() */
+/* Flags passed to regcomp() */
 int		 cflags = REG_NOSUB | REG_NEWLINE;
-int		 eflags = REG_STARTEND;
```

Comment:
This is a very odd change.... It should be #ifdef linux or something, no?

Reply:
There's `#ifdef REG_STARTEND` but inside of `util.c`. There it's choosing between using the flag if it exists, and a fallback manual implementation in the `#else`. With that, the global default for the flags doesn't quite make sense.